dashboard/app/models/user_school_info.rb

```diff
@@ -0,0 +1,18 @@
+# == Schema Information
+#
+# Table name: user_school_infos
+#
+#  id                     :integer          not null, primary key
+#  user_id                :integer          not null
+#  start_date             :datetime         not null
+#  end_date               :datetime
+#  school_info_id         :integer          not null
+#  last_confirmation_date :datetime         not null
+#  created_at             :datetime         not null
+#  updated_at             :datetime         not null
+#
+
+class UserSchoolInfo < ApplicationRecord
+  validates_presence_of :user
```

Comment:
you'll want to validate the presence of any column that cannot be null here, not just user and school_info:

```
validates_presence_of :user, :school_info, :start_date, :last_confirmation_date
```

Reply:
Stumbled upon this looking for something else...out of curiosity, what will happen here if a record is attempted to be added without `:school_info, :start_date, :last_confirmation_date`?

Reply:
@bencodeorg Good question.  The record will not be added and a database error will be thrown.  I checked the users table and see that null is acceptable for the following columns :school_info, :start_date, :last_confirmation_date columns.  

In addition, I can also create a new teacher account without entering school information.  Do we want the user_school_infos table to have the same set up?

Reply:
I think your set up is correct -- having a null `school_info_id` in `users` is still ok (if a teacher didn't provide any school information), but we wouldn't want any records in this table to have a null `school_info_id`.

Reply:
If `:school_info, :start_date, :last_confirmation_date` can all be null, you will need to add another migration that makes those columns `null: true`. Right now, the database is requiring those fields via `null: false`

Reply:
I talked to Ben and I redact my last comment! Ignore me. @nkiruka can you create a second PR that adds the missing validations to our `UserSchoolInfo` model? We're still missing validations for `start_date` and `last_confirmation_date`